### PR TITLE
corp: support limit and reverse scan

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @qiuyesuifeng @coocood  PTAL

### CODE REVIEW

File: src/server/coprocessor.rs [hunk 1]
```diff
@@ -395,32 +429,50 @@ impl<'a> SelectContext<'a> {
...
             if nk.is_empty() {
                 debug!("no more data to scan");
-                return Ok(rows);
+                break;
             }
             let (key, value) = try!(nk.pop().unwrap());
-            if r.get_end() <= &key {
-                debug!("reach end key: {} >= {}", escape(&key), escape(r.get_end()));
-                return Ok(rows);
+            if desc && r.get_start() > &key || !desc && r.get_end() <= &key {
```
Comment (coocood) [inline src/server/coprocessor.rs]: The if condition is too complex.

File: src/server/coprocessor.rs [hunk 2]
```diff
@@ -290,55 +300,79 @@ impl<'a> SelectContext<'a> {
...
+                };
                 trace!("scan takes {:?}", timer.elapsed());
                 if res.is_empty() {
                     debug!("no more data to scan.");
                     break;
                 }
                 let (key, _) = try!(res.pop().unwrap());
-                if range.get_end() <= &key {
-                    debug!("reach end key: {} >= {}",
+                if desc && range.get_start() > &key || !desc && range.get_end() <= &key {
```
Comment (coocood) [inline src/server/coprocessor.rs]: The if condition is too complex.

**Comment by BusyJay** [comment]
PTAL

**Comment by coocood** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**